rosbag2/src/rosbag2/impl/sequential_reader_impl.hpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2__IMPL__SEQUENTIAL_READER_IMPL_HPP_
+#define ROSBAG2__IMPL__SEQUENTIAL_READER_IMPL_HPP_
+
+#include "rosbag2/sequential_reader.hpp"
+
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace rosbag2
+{
+
+class SequentialReader::ReaderImpl
```

Comment:
(minor) it'd be preferable to have a cpp file for this one too

Reply:
Sure, keep under `src/rosbag2/impl` or just `src/rosbag2`?

Reply:
everything in src is impl :)
I'd just use src/rosbag2